DOCS-6510 Changes (#16)
mulesoft/docs-functional-monitoring: modules/ROOT/pages/bat-start-new-project.adoc

= Start a New Test Suite
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Starting a new test suite for testing API endpoints involves only a few steps to create and populate with initial lines a file named `bat.yaml`. This file explains the structure of your test suite to the BAT CLI.

> == To Start a New Test Suite

Reviewer: You don't need this subhead.

. Create a directory in your computer's filesystem.
. Change to the new directory.
. Create a file named `bat.yaml`.
. Open the file for editing.

> . Add these five lines:

Reviewer: ```suggestion
. Add these five lines to specify the name that you want to give the suite of tests that you plan to write:
```

+
[source,yaml]
----
suite:
  name: "name_of_suite_of_tests"

files:

----
+

> name_of_suite_of_tests:: Specify the name that you want to give the suite of tests that you plan to write.

Reviewer: omit
```suggestion
```

. Save the file.

> . Create a folder in your directory in which to create tests. You can give the folder any name.

Reviewer: ```suggestion
. Create a folder in your directory in which to create tests. 
You can give the folder any name.
```

> . Create a folder named `config`. In this folder, you create configuration files to use for your tests. Configuration files contain information that the BAT CLI needs to connect to the systems where the APIs that you want to test are located.

Reviewer 2: Optionally create the config folder.
Also I would change the last sentence to something like this: `Configuration files contain variables that are injected to tests in runtime. You can refer to those variables int tests using the config variable: '$(config.url)'.`
And when you execute from the CLI you can specify which configuration file to load variables from `--config=configFile.dwl`, and from the UI when you set up the information for the schedule.

Author: Resolved.

> . Optional: Create a folder named `config`. In this folder, you create configuration files to use for your tests. Configuration files contain variables that are injected to tests at runtime. You can refer to those variables in tests by using the `config` variable. For example, to refer to a variable named `url` in a configuration file, you use this reference: '$(config.url)'. When you run the `bat` command to execute a test, you specify the configuration file to use by including the `--config=file.dwl` option, where `file.dwl` is the path and name of the configuration file in the folder for your test suite.

Reviewer: ```suggestion
. Optionally, create a folder named `config` in which you create configuration files to use for your tests.

Configuration files contain variables that are injected to tests at runtime. You can refer to those variables in tests by using the `config` variable. 

For example, to refer to a variable named `url` in a configuration file, you use this reference: '$(config.url)'. When you run the `bat` command to execute a test, you specify the configuration file to use by including the `--config=file.dwl` option, where `file.dwl` is the path and name of the configuration file in the folder for your test suite.
```

== What to Do Next

> Now, you can start writing tests.

Reviewer: I recommend not using "now" in chronological content, in favor of a more precise context such as "After/Before <you do this/have done this> or <this has happened/this can happen >" ...."
Or "next."
